data_model/src/domain.rs

```diff
@@ -82,39 +145,95 @@ impl Ord for Domain {
 }
 
 impl Domain {
-    /// Construct [`Domain`] from [`Id`].
-    pub fn new(id: Id) -> Self {
-        Self {
-            id,
-            accounts: AccountsMap::new(),
-            asset_definitions: AssetDefinitionsMap::new(),
-            metadata: Metadata::new(),
-            logo: None,
-        }
+    /// Construct new empty [`Domain`] from [`Id`].
+    pub fn new(id: Id) -> NewDomain {
+        NewDomain::new(id)
     }
 
-    /// Domain constructor with pre-setup accounts. Useful for testing purposes.
-    pub fn with_accounts(name: &str, accounts: impl IntoIterator<Item = Account>) -> Self {
-        let accounts_map = accounts
-            .into_iter()
-            .map(|account| (account.id.clone(), account))
-            .collect();
-        #[allow(clippy::expect_used)]
-        Self {
-            id: Id::new(name).expect("must be valid"),
-            accounts: accounts_map,
-            asset_definitions: AssetDefinitionsMap::new(),
-            metadata: Metadata::new(),
-            logo: None,
-        }
+    /// Returns a reference to the account corresponding to the account id.
```

Comment:
```suggestion
    /// Return a reference to the account corresponding to the account id.
```
Minor thing, but function definitions in imperative is part of our `Contributing.md`.